library/Backends/Plonk/src/Juvix/Backends/Plonk/Parameterization.hs

```diff
@@ -0,0 +1,238 @@
+{-# LANGUAGE UndecidableInstances #-}
+{-# OPTIONS_GHC -Wwarn=incomplete-patterns #-}
+
+module Juvix.Backends.Plonk.Parameterization
+  ( module Juvix.Backends.Plonk.Parameterization,
+    module Types,
+  )
+where
+
+import Data.Field.Galois (GaloisField (..))
+import qualified Data.List.NonEmpty as NonEmpty
+import Juvix.Backends.Plonk.Types as Types
+import qualified Juvix.Core.Application as App
+import qualified Juvix.Core.ErasedAnn.Prim as Prim
+import qualified Juvix.Core.ErasedAnn.Types as ErasedAnn
+import qualified Juvix.Core.IR.Evaluator as Eval
+import qualified Juvix.Core.IR.Types.Base as IR
+import qualified Juvix.Core.Parameterisation as Param
+import qualified Juvix.Core.Types as Core
+import Juvix.Library hiding (many, show, try)
+import qualified Juvix.Library.HashMap as Map
+import qualified Juvix.Library.NameSymbol as NameSymbol
+import qualified Juvix.Library.Usage as Usage
+import Prelude (Show (..))
+
+check3Equal :: Eq a => NonEmpty a -> Bool
```

Comment:
I think some of these can be moved into code common to all parameterisations or a stdlib, this can happen later though